webpack.config.js

```diff
@@ -49,262 +52,265 @@ if ( defaultConfig.module && Array.isArray( defaultConfig.module.rules ) ) {
+		config.entry = Array.isArray( asset.entry )
+			? {
+					[ asset.chunk ]: asset.entry,
+			  }
+			: asset.entry;
 
-const overrides = {
-	...defaultConfig,
-	...sharedConfig,
-	entry: {
-		overrides: [ './assets/css/src/overrides.css' ],
-	},
-	plugins: [
-		...sharedConfig.plugins,
-		new FixStyleOnlyEntriesPlugin(),
-		new WebpackBar( {
-			name: 'Overrides',
-			color: '#a81773',
-		} ),
-	],
-};
+		config.plugins = [
+			...config.plugins,
+			new WebpackBar( {
+				name: `${ packageType }: ${ asset.name }`,
+				color: randomColor(),
+			} ),
+		];
 
-const wizard = {
-	...defaultConfig,
-	...sharedConfig,
-	entry: {
-		wizard: [ './assets/src/wizard/index.js', './assets/css/src/wizard.css' ],
-	},
-	plugins: [
-		...sharedConfig.plugins,
-		new WebpackBar( {
-			name: 'wizard',
-			color: '#707a8a', // 🧙🏼‍♂️
-		} ),
-	],
-};
+		if (
+			asset.hasOwnProperty( 'toBundle' ) &&
+			Array.isArray( asset.toBundle )
+		) {
+			config.plugins = [
+				...config.plugins.filter(
+					// Remove the `DependencyExtractionWebpackPlugin` if it already exists.
+					plugin => ! ( plugin instanceof DependencyExtractionWebpackPlugin )
+				),
+				new DependencyExtractionWebpackPlugin( {
+					useDefaults: false,
+					requestToExternal( request ) {
+						if ( asset.toBundle.includes( request ) ) {
+							return undefined;
+						}
 
-// Getting Started Module.
-const gsm = {
-	...defaultConfig,
-	...sharedConfig,
-	entry: {
-		'getting-started': [
-			'./assets/src/getting-started/index.js',
-			'./assets/css/src/getting-started.css',
-		],
-	},
-	plugins: [
-		...sharedConfig.plugins,
-		new WebpackBar( {
-			name: 'Getting Started',
-			color: '#3ce1bb',
-		} ),
-	],
-};
+						return defaultRequestToExternal( request );
+					},
+					requestToHandle( request ) {
+						if ( asset.toBundle.includes( request ) ) {
+							return 'wp-block-editor'; // Return block-editor as a dep.
+						}
 
-// These packages need to be bundled and not extracted to `wp.*`.
-const PACKAGES_TO_BUNDLE = [
-	'@wordpress/data',
-	'@wordpress/escape-html',
-	'@wordpress/rich-text',
-];
+						return defaultRequestToHandle( request );
+					},
+				} ),
+			];
+		}
 
-const polyfills = {
-	...defaultConfig,
-	...sharedConfig,
-	entry: {
-		polyfills: [ './assets/src/polyfills/index.js' ],
-	},
-	plugins: [
-		...sharedConfig.plugins.filter(
-			// Remove the `DependencyExtractionWebpackPlugin` if it already exists.
-			plugin => ! ( plugin instanceof DependencyExtractionWebpackPlugin )
-		),
-		new DependencyExtractionWebpackPlugin( {
-			useDefaults: false,
-			requestToExternal( request ) {
-				if ( PACKAGES_TO_BUNDLE.includes( request ) ) {
-					return undefined;
-				}
+		if ( asset.chunk === 'block-editor' ) {
+			config.plugins = [
+				...config.plugins,
+				new CopyWebpackPlugin( {
+					patterns: [
+						{
+							from: './plugin/assets/src/block-editor/blocks/*/block.json',
+							to: './blocks/[1]/block.json',
```

Comment:
Is the `./blocks/[1]/block.json` correct? Is it relative to the plugin directory by default?

Reply:
This is relative to the plugin path, the `path` value [here](https://github.com/xwp/material-design-wp-plugin/blob/e0f797fce7de8aea124068293c17fea74401fe77/webpack.config.js#L163) resolves the correct dest path based on the `packageType` (plugin or theme).